src/Tribe/Template.php

```diff
@@ -502,13 +502,27 @@ public function get_template_file( $name ) {
 	 * @return string|false Either the final content HTML or `false` if no template could be found.
 	 */
 	public function template( $name, $context = array(), $echo = true ) {
-		// If name is String make it an Array
-		if ( is_string( $name ) ) {
-			$name = (array) explode( '/', $name );
+		static $file_exists    = [];
+		static $files          = [];
+		static $template_names = [];
+
+		// Key we'll use for in-memory caching of expensive operations.
+		$cache_name_key = $name;
+
+		// Cache template name massaging so we don't have to repeat these actions.
+		if ( ! isset( $template_names[ $cache_name_key ] ) ) {
+			// If name is String make it an Array
```

Comment:
:no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: 046083aee4925c0622830e7c399a8d93</sub>